managed/src/main/java/com/yugabyte/yw/models/helpers/CustomerConfigValidator.java

```diff
@@ -61,6 +72,10 @@ public void validate(String type, String name, JsonNode data, ObjectNode errorsC
       if (this.type.equals(type) && this.name.equals(name)) {
         JsonNode value = data.get(fieldName);
         doValidate(value == null ? "" : value.asText(), errorsCollector);
+        if (this.name.equals("S3") &&
```

Comment:
Can't understand the reason why it is here. You can create one mare validating class and add it to `validators`. Already inside it you can check whether "AWS_ACCESS_KEY_ID" is present or not and to do the checks.

Reply:
Thats what I had suggested in my original suggestion.

Reply:
I thought of going that route but will have to create another nested class and extends ConfigValidator and do the validations there. 
This sounds ok now. Will do this changes. Thanks!

In general, for all the API's can we have some kind of design patterns implemented in place may be TemplateMethod design pattern where we will have 

Abstract class AbstractCommand {

 public void execute() {
  prepareToExecute(); - for all the input validations
  doExecute(); - to process the inputs
  doGetResults(); - format the result
  }

@sb-yb @SergeyPotachev  Any thoughts of above? Thanks!

Reply:
There is already a way to do this sort of separation cleanly in play framework. We do not use that as of now. But we can. One obstacle was lack of exception handler. Now that we have that may be we can simplify this further. 
But that is independent. What @SergeyPotachev has added here is plugging in small validation logic specific to a single field and easily add data driven unit tests for that. That will be useful no matter what's the high level invocation of validation logic pans out.